src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -690,18 +706,83 @@ public void decrementSelectedCount()
             String selectedUpdate = createSelectedUpdateMessage(false);
             if (logger.isDebugEnabled())
             {
-                logger.debug("Endpoint " + getID() + " is no longer "
-                    + "selected, sending message: " + selectedUpdate);
+                logger.debug(logPrefix +
+                        "Is no longer selected, sending message: " +
+                        selectedUpdate);
             }
             try
             {
                 sendMessage(selectedUpdate);
             }
             catch (IOException e)
             {
-                logger.error("Error sending SelectedUpdate message: " + e);
+                logger.error(logPrefix +
+                        "Error sending SelectedUpdate message: " + e);
+            }
+        }
+    }
+
+    /**
+     * Gets this {@link Endpoint}'s transport manager. If there was a previous
+     * attempt to initialize the transport manager which failed, does not
+     * re-try and returns {@code null}.
+     *
+     * @return this {@link Endpoint}'s transport manager, or {@code null} if it
+     * failed to initialize.
+     */
+    public IceDtlsTransportManager getTransportManager()
+    {
+        if (transportManager != null)
+        {
+            return transportManager;
+        }
+        else if (transportManagerException != null)
+        {
+            // We've already tried and failed to initialize the TM.
```

Comment:
should we just rethrow it here? (and throw it down below)?  we could still save it, but re-throwing looks like it'd make the logic below in `setTransportInfo` easier (instead of checking for a null return and then checking if the exception member is set, we'd just catch the exception)

Reply:
Yeah, I'll do that. I'm not a fan of getters that throw exceptions, but with the current usage it will be better this way